src/api/objects/ObjectAPI.js

```diff
@@ -184,6 +184,15 @@ ObjectAPI.prototype.get = function (identifier, abortSignal) {
     }
 
     identifier = utils.parseKeyString(identifier);
+    let dirtyObject;
+    if (this.isTransactionActive()) {
+        dirtyObject = this.transaction.getDirtyObject(keystring);
```

Comment:
We already have this object, so why are we getting it again?

There _may_ be a legitimate reason that `get` is being called again for objects that we already have in memory, but I'd like to understand what that is, because we may be treating a symptom and not the cause.

Reply:
yeah its actually triggered by `Selection` 
https://github.com/nasa/openmct/blob/933ce7aa3c01643b8f7254ea9ce15b0c94a881bc/src/selection/Selection.js#L204